### DISCUSSION TITLE
How to "re-link" my fork to its "grandparent" repo instead of "parent" fork?

### DISCUSSION THREAD

**Discussion by penyuan** [community/community#22103] [New to GitHub]
<p>Sorry I don’t know the correct terminology: Suppose there is a git repository by Alice, and Bob made a fork of it. Next, I made a fork of Bob’s fork. So the fork “lineage” is:</p>
<pre><code class="lang-auto">Alice -&gt; Bob's fork -&gt; my fork
</code></pre>
<p>Is there a way to change the plumbing so that my fork directly tracks Alice’s original repo instead of Bob’s fork? Like this:</p>
<pre><code class="lang-auto">Alice -&gt; my fork
</code></pre>
<p>This way, I can choose to incorporate changes in Alice’s repo directly into my fork instead of waiting for Bob to incorporate them into his.</p>
<p>I know forking is more of a GitHub instead of a pure git concept, but if there’s a way to do this for generic git repositories I’d like to know, too! Thanks!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Having the forks restructured would not give you automatic updates. The generic solution is:</p>
<ol>
<li>Add Alice’s repo to your <em>local</em> repo as a remote (<code>git remote add</code>).</li>
<li>Fetch updates from that remote (<code>git fetch</code>).</li>
<li>Create (or update) a dedicated local branch (<code>upstream</code> is a common name for that). If you don’t touch that branch otherwise, you should be able to fast-forward it on updates, unless Alice does a rebase.</li>
<li>Push that branch to your Github repo.</li>
</ol>

**Comment by penyuan** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="122760">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>The generic solution is:</p>
</blockquote>
</aside>
<p>Ah that makes sense, thank you so much! :heart:</p>
<p>Just to confirm, would this mean that it would solely be my <em>local</em> repository’s <code>upstream</code> branch that would be tracking Alice’s repository? I.e. as far as GitHub is concerned, my fork repository hosted on GitHub is still linked to Bob’s fork and GitHub wouldn’t be “aware” that my <code>upstream</code> branch is tracking Alice? In other words, I would be tracking Alice locally via my local <code>upstream</code> branch (with Alice’s repo set as <code>upstream</code>'s remote) and not rely on GitHub at all for that purpose, right?</p>

**Comment by airtower-luna** [comment]
<p>Yes, exactly. :slightly_smiling_face:</p>

**Comment by penyuan** [comment]
<p>Makes sense. I will use this solution for my forks in this situation. Thank you so much! :grin: :clap:</p>

**Comment by tony-p** [comment]
To reroute on the github side you need to open a support ticket (so it appears as a fork of the upstream repo) 

there is a dedicated subject for it.

https://support.github.com/contact?legacy&subject=Reroute%20a%20Fork&tags=rr-forks

**Reply by SSasindu** [reply]
I have opened a reroute request from github support. Because I have accidentally change parent repo visibility from private to public, without knowing it will affect the fork relationship. Can they fix that? If yes, how long it will take according to your experience, if you have posted that type issue to them?

**Reply by tony-p** [reply]
Reroute took about 1 working day.
